src/executables/util_cs.c

```diff
@@ -2727,6 +2746,104 @@ changemode (UTIL_FUNCTION_ARG * arg)
 #endif /* !WINDOWS */
 }
 
+int
+rkcheck (UTIL_FUNCTION_ARG * arg)
+{
+  int error;
+  DB_OBJLIST *classes, *c;
+  UTIL_ARG_MAP *arg_map = arg->arg_map;
+  char *database_name, *user, *password = NULL;
+  int violation_count = 0;
+  char **dbs;
+  char er_msg_file[PATH_MAX];
+  char violation_list_file[PATH_MAX];
+  char violation_list_file_path[PATH_MAX];
+  FILE *fp;
+  int ret;
+
+  database_name = utility_get_option_string_value (arg_map, OPTION_STRING_TABLE, 0);
+  user = utility_get_option_string_value (arg_map, UNLOAD_USER_S, 0);
+  password = utility_get_option_string_value (arg_map, UNLOAD_PASSWORD_S, 0);
+
+  /* error message log file */
+  snprintf (er_msg_file, sizeof (er_msg_file) - 1, "%s_%s.err", database_name, arg->command_name);
+  er_init (er_msg_file, ER_NEVER_EXIT);
+  er_set_print_property (ER_PRINT_TO_CONSOLE);
+
+  snprintf (violation_list_file, sizeof (violation_list_file) - 1, "%s_%s.list", database_name, arg->command_name);
+  envvar_logdir_file (violation_list_file_path, PATH_MAX, violation_list_file);
+  violation_list_file_path[PATH_MAX - 1] = '\0';
+  fp = fopen (violation_list_file_path, "w");
+
+  if (user == NULL || user[0] == '\0')
+    {
+      user = (char *) "DBA";
+    }
+
+  error = db_restart_ex (arg->command_name, database_name, user, password, NULL, DB_CLIENT_TYPE_ADMIN_UTILITY);
+  if (error != NO_ERROR)
+    {
+      fclose (fp);
+      // TODO: Check if password-error handling (like unloaddb) is needed, even though this logic doesn’t use authentication.
+      PRINT_AND_LOG_ERR_MSG ("%s: %s\n", arg->command_name, db_error_string (3));
+      return ER_FAILED;
+    }
+
+  classes = db_get_all_classes ();
+  if (classes == NULL)
+    {
+      fclose (fp);
+      PRINT_AND_LOG_ERR_MSG ("%s: %s\n", arg->command_name, db_error_string (3));
+      return ER_FAILED;
+    }
+
+  PRINT_SECTION_TITLE (fp, RK_CONSTRAINT_VIOLATIONS_SECTION_TITLE);
+  for (c = classes; c != NULL; c = c->next)
+    {
+      if (db_is_system_class (c->op) || !sm_is_replication_class (c->op))
+	{
+	  continue;
+	}
+
+      if (!classobj_has_class_repl_key_constraint (db_get_constraints (c->op)))
+	{
+	  fprintf (fp, "%s\n", sm_get_ch_name (c->op));
+	  violation_count++;
+	}
+    }
+
+  PRINT_SECTION_BLACK (fp);
+  PRINT_SECTION_TITLE (fp, FK_CONSTRAINT_VIOLATIONS_SECTION_TITLE);
+  for (c = classes; c != NULL; c = c->next)
+    {
+      if (db_is_system_class (c->op) || !sm_is_replication_class (c->op))
+	{
+	  continue;
+	}
+      ret = log_ha_repl_fk_ref_all_replicated (c->op, fp);
+      if (ret > 0)
+	{
+	  violation_count += ret;
+	}
+    }
+
+  if (classes)
+    {
+      db_objlist_free (classes);
+    }
+
+  fclose (fp);
+
+  if (violation_count > 0)
+    {
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_HA_REPLICATION_CONSTRAINT_VIOLATION, 2, violation_count,
+	      er_msg_file);
```

Comment:
PR에 설명하신 대로 구현은 되어 있지만, '.list 파일'을 참조하라고 출력하는 것을 의도하신게 아닌가 생각됩니다.
```suggestion
	      violation_list_file);
```

Reply:
@hornetmj 잘못 작성된 변수명이라 violation_list_file로 변경했습니다.